controllers/core/openstackcontrolplane_controller.go

```diff
@@ -170,10 +172,81 @@ func (r *OpenStackControlPlaneReconciler) Reconcile(ctx context.Context, req ctr
 		return ctrl.Result{}, nil
 	}
 
-	return r.reconcileNormal(ctx, instance, helper)
+	Log.Info("Looking up the current OpenStackVersion")
+	ctrlResult, err, version := openstack.ReconcileVersion(ctx, instance, helper)
+	if err != nil {
+		return ctrl.Result{}, err
+	} else if (ctrlResult != ctrl.Result{}) {
+		return ctrlResult, nil
+	}
+
+	// wait until the version is initialized so we have images on the version.Status
+	if version.Status.Conditions.IsFalse(corev1beta1.OpenStackVersionInitialized) {
+		return ctrlResult, nil
+	}
+
+	if instance.Status.DeployedVersion == nil || version.Spec.TargetVersion == *instance.Status.DeployedVersion {
+		// green field deployment or no minor update in progress
+		ctrlResult, err := r.reconcileNormal(ctx, instance, version, helper)
+		if err != nil {
+			Log.Info("Error reconciling normal", "error", err)
+			return ctrl.Result{}, err
+		} else if (ctrlResult != ctrl.Result{}) {
+			Log.Info("Reconciling normal")
+			return ctrlResult, nil
+		}
+		// this will allow reconcileNormal to proceed in subsequent reconciles
+		instance.Status.DeployedVersion = &version.Spec.TargetVersion
+		return ctrl.Result{}, nil
+	} else {
+		if version.Status.Conditions.IsFalse(corev1beta1.OpenStackVersionMinorUpdateOVNControlplane) {
+			Log.Info("Minor update OVN on the ControlPlane")
+			ctrlResult, err := r.reconcileOVNControllers(ctx, instance, version, helper)
+			if err != nil {
+				return ctrl.Result{}, err
+			} else if (ctrlResult != ctrl.Result{}) {
+				return ctrlResult, nil
+			}
+			instance.Status.DeployedOVNVersion = &version.Spec.TargetVersion
+			return ctrl.Result{}, nil
+		} else if version.Status.Conditions.IsFalse(corev1beta1.OpenStackVersionMinorUpdateControlplane) {
+			Log.Info("Minor update on the ControlPlane")
+			ctrlResult, err := r.reconcileNormal(ctx, instance, version, helper)
+			if err != nil {
+				return ctrl.Result{}, err
+			} else if (ctrlResult != ctrl.Result{}) {
+				return ctrlResult, nil
+			}
+			// this will allow reconcileNormal to proceed in subsequent reconciles
+			instance.Status.DeployedVersion = &version.Spec.TargetVersion
+			return ctrl.Result{}, nil
+		} else {
+			Log.Info("Skipping reconcile. Waiting on minor update to proceed.")
```

Comment:
I think the message here is misleading. At this point both the ovn and the control plane update is ready. I'm not even sure when we hit this branch. If both are ready then I expect that when the control plane finished it already set the DeployedVersion in the status so we should not end up here but in the normal reconcile part in the if branch.

Maybe this is the result of the bug that the else if branches are not handling the nil or Unknown condition states?